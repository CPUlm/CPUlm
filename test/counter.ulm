.text
; Ce programme incrémente r3 jusqu'à ce que r3 = max

main:
    loadi   r2  $max
    load    r2  r2
    loadi   r3  0

loop:
    sub     r4  r2  r3
    jmp.z   $end
    inc     r3  r3
    jmp     $loop

end:
    halt

.data
max: .int 5
