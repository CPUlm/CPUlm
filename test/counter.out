
Step: 1

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000001 |         0x1 |                    1 |                    1 |
=>  r2               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r3               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r4               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 2

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000010 |         0x2 |                    2 |                    2 |
=>  r2               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r3               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r4               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 3

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000011 |         0x3 |                    3 |                    3 |
=>  r2               | 0b00000000000000000001000000000000 |      0x1000 |                 4096 |                 4096 |
=>  r3               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r4               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 4

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000100 |         0x4 |                    4 |                    4 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r4               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 5

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r4               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 6

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000110 |         0x6 |                    6 |                    6 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r4               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 7

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000111 |         0x7 |                    7 |                    7 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r4               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 8

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000001000 |         0x8 |                    8 |                    8 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000001 |         0x1 |                    1 |                    1 |
=>  r4               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 9

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000001 |         0x1 |                    1 |                    1 |
=>  r4               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 10

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000110 |         0x6 |                    6 |                    6 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000001 |         0x1 |                    1 |                    1 |
=>  r4               | 0b00000000000000000000000000000100 |         0x4 |                    4 |                    4 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 11

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000111 |         0x7 |                    7 |                    7 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000001 |         0x1 |                    1 |                    1 |
=>  r4               | 0b00000000000000000000000000000100 |         0x4 |                    4 |                    4 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 12

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000001000 |         0x8 |                    8 |                    8 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000010 |         0x2 |                    2 |                    2 |
=>  r4               | 0b00000000000000000000000000000100 |         0x4 |                    4 |                    4 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 13

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000010 |         0x2 |                    2 |                    2 |
=>  r4               | 0b00000000000000000000000000000100 |         0x4 |                    4 |                    4 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 14

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000110 |         0x6 |                    6 |                    6 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000010 |         0x2 |                    2 |                    2 |
=>  r4               | 0b00000000000000000000000000000011 |         0x3 |                    3 |                    3 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 15

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000111 |         0x7 |                    7 |                    7 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000010 |         0x2 |                    2 |                    2 |
=>  r4               | 0b00000000000000000000000000000011 |         0x3 |                    3 |                    3 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 16

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000001000 |         0x8 |                    8 |                    8 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000011 |         0x3 |                    3 |                    3 |
=>  r4               | 0b00000000000000000000000000000011 |         0x3 |                    3 |                    3 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 17

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000011 |         0x3 |                    3 |                    3 |
=>  r4               | 0b00000000000000000000000000000011 |         0x3 |                    3 |                    3 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 18

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000110 |         0x6 |                    6 |                    6 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000011 |         0x3 |                    3 |                    3 |
=>  r4               | 0b00000000000000000000000000000010 |         0x2 |                    2 |                    2 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 19

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000111 |         0x7 |                    7 |                    7 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000011 |         0x3 |                    3 |                    3 |
=>  r4               | 0b00000000000000000000000000000010 |         0x2 |                    2 |                    2 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 20

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000001000 |         0x8 |                    8 |                    8 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000100 |         0x4 |                    4 |                    4 |
=>  r4               | 0b00000000000000000000000000000010 |         0x2 |                    2 |                    2 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 21

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000100 |         0x4 |                    4 |                    4 |
=>  r4               | 0b00000000000000000000000000000010 |         0x2 |                    2 |                    2 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 22

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000110 |         0x6 |                    6 |                    6 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000100 |         0x4 |                    4 |                    4 |
=>  r4               | 0b00000000000000000000000000000001 |         0x1 |                    1 |                    1 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 23

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000111 |         0x7 |                    7 |                    7 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000100 |         0x4 |                    4 |                    4 |
=>  r4               | 0b00000000000000000000000000000001 |         0x1 |                    1 |                    1 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 24

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000001000 |         0x8 |                    8 |                    8 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r4               | 0b00000000000000000000000000000001 |         0x1 |                    1 |                    1 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 25

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r4               | 0b00000000000000000000000000000001 |         0x1 |                    1 |                    1 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 26

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000110 |         0x6 |                    6 |                    6 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r4               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 27

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000001001 |         0x9 |                    9 |                    9 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r4               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |

Step: 28

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000001010 |         0xa |                   10 |                   10 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r4               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b00000000000000001111111111111111 |      0xffff |                65535 |                65535 |

Step: 29

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000001011 |         0xb |                   11 |                   11 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r4               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b11111111111111111111111111111111 |  0xffffffff |           4294967295 |                   -1 |

Step: 30

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b11111111111111111111111111111111 |  0xffffffff |           4294967295 |                   -1 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r4               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b11111111111111111111111111111111 |  0xffffffff |           4294967295 |                   -1 |

Step: 31

       Outputs       |               Binary               | Hexadecimal |    Signed Decimal    |   Unsigned Decimal   |
---------------------+------------------------------------+-------------+----------------------+----------------------+
=>  pc               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r2               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r3               | 0b00000000000000000000000000000101 |         0x5 |                    5 |                    5 |
=>  r4               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r5               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  r6               | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rout             | 0b00000000000000000000000000000000 |           0 |                    0 |                    0 |
=>  rpriv            | 0b11111111111111111111111111111111 |  0xffffffff |           4294967295 |                   -1 |
